org.jacoco.core/src/org/jacoco/core/internal/flow/Instruction.java

```diff
@@ -102,11 +121,11 @@ public int getBranches() {
 	}
 
 	/**
-	 * Returns the number of covered branches starting from this instruction.
-	 * 
-	 * @return number of covered branches
+	 * Returns coverage of branches of this instruction.
+	 *
+	 * @return coverage of branches of this instruction
 	 */
-	public int getCoveredBranches() {
+	public BitSet getCoveredBranches() {
```

Comment:
> allows internal optimizations later

I suppose internal in respect to a class and not project since class is already in internal package 😉 

Jokes aside - while indeed such encapsulation looks cleaner, this will complicate a test a bit which currently able to directly assert on a content of this `BitSet`, but ok can do this.